packages/modal/src/Modal.js

```diff
@@ -6,6 +6,35 @@ import { color, width, height } from 'styled-system'
 import { DialogOverlay, DialogContent } from '@reach/dialog'
 import { Box, CloseButton, Flex } from 'pcln-design-system'
 
+const OVERLAY_ANIMATION = transitionstate => `
+  opacity: 0;
+  transition: opacity .5s cubic-bezier(0.50, 0.00, 0.25, 1.00);
+  ${transitionstate === 'entering' ? `opacity: 0;` : ''}
+  ${transitionstate === 'entered' ? `opacity: 1;` : ''}
+  ${transitionstate === 'exiting' ? `opacity: 0;` : ''}
+  ${transitionstate === 'exited' ? `opacity: 0;` : ''}
+`
+
+const DIALOG_ANIMATION = transitionstate => `
+  transform: scale(0.5);
+  transition: transform .5s cubic-bezier(0.50, 0.00, 0.25, 1.00);
+  ${transitionstate === 'entering' ? `transform: scale(0.5);` : ''}
+  ${transitionstate === 'entered' ? `transform: scale(1);` : ''}
+  ${transitionstate === 'exiting' ? `transform: scale(0.5);` : ''}
+  ${transitionstate === 'exited' ? `transform: scale(0.5);` : ''}
+`
+
+const getAnimation = ({
+  transitionstate,
+  defaultAnimation,
+  customAnimation = null
+}) => {
+  if (customAnimation && typeof customAnimation === 'function') {
+    return customAnimation(transitionstate)
+  }
+  return defaultAnimation(transitionstate)
+}
```

Comment:
Could be simplified due to `typeof` handles nullity check already, and a `return` statement with a ternary expression will omit the `if` imperative block. Also camel case can be consistent for param `transitionState`:

```js
const getAnimation = ({
  transitionState,
  defaultAnimation,
  customAnimation = null
}) => typeof customAnimation === 'function'
  ? customAnimation(transitionstate)
  : defaultAnimation(transitionstate)
```